stm32-modules/thermocycler-refresh/src/colors.cpp

```diff
@@ -0,0 +1,47 @@
+/**
+ * @file colors.cpp
+ * @brief Implements fetching color structures from codes
+ */
+
+#include "thermocycler-refresh/colors.hpp"
+
+using namespace colors;
+
+static constexpr uint8_t FULL = 0xFF;
+static constexpr uint8_t HIGH = 0xEE;
+static constexpr uint8_t MED = 0x50;
+
+// Public function implementation
+auto colors::get_color(Colors color, double brightness) -> xt1511::XT1511 {
```

Comment:
I think we're going to end up having a fixed color palette that gets fine tuned in implementation to look correct; I think we should get ahead of the game a bit here by just specifying values for the four color channels. We should also establish and document an explicit color depth we support.